### PR TITLE
migrating view_pools for curvefi on ethereum to spellbook

### PR DESCRIPTION
**Pull request by agaperste**
Brief comments on the purpose of your changes:
Migrating existing abstraction curvefi."view" to spellbook.

*For Dune Engine V2*
I've checked that:

* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`

@duneanalytics/team-data-experience
Please let me know how this looks!

### CODE REVIEW

File: dbt_project.yml [hunk 1]
```diff
@@ -136,6 +136,13 @@ models:
         +schema: safe_ethereum
         +materialized: view
 
+    curvefi:
+      +schema: safe
```
Comment (jeff-dude) [inline dbt_project.yml:140]: imagine this is a copy/paste error and should be curvefi?

File: models/curvefi/ethereum/curvefi_ethereum_sources.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        loaded_at_field: call_block_time
+      - name: CurveFactoryV2_evt_CryptoPoolDeployed
+        loaded_at_field: evt_block_time
+      - name: CurveFactoryV2_call_deploy_pool
+        loaded_at_field: call_block_time
+      - name: CurveFactory_evt_LiquidityGaugeDeployed
+        loaded_at_field: evt_block_time
+      - name: CurveFactoryV2_evt_LiquidityGaugeDeployed
+        loaded_at_field: evt_block_time
+  - name: ethereum
```
Comment (jeff-dude) [inline models/curvefi/ethereum/curvefi_ethereum_sources.yml:24]: i believe this source would fit into the 'base_sources' model as it's on ethereum schema, not curvefi specific

File: models/curvefi/ethereum/curvefi_ethereum_schema.yml [hunk 3]
```diff
@@ -0,0 +1,85 @@
+version: 2
+
+models:
+  - name: curvefi_ethereum_view_pools
+    meta:
+      blockchain: ethereum
+      project: curvefi
+      contributors: [agaperste, yulesa]
+    config:
+      tags: ["ethereum", "curvefi", "view_pools"]
```
Comment (jeff-dude) [inline models/curvefi/ethereum/curvefi_ethereum_schema.yml:10]: can you add 'view' as a tag? we're thinking through our tagging strategy, so there may be tweaks in another future PR, but tags will help us with job orchestration downstream for automation

File: dbt_project.yml [hunk 4]
```diff
@@ -1,11 +1,11 @@
 #Welcome to your Spellbook!
 
-name: 'spellbook'
-version: '1.0.0'
+name: "spellbook"
```
Comment (jeff-dude) [inline dbt_project.yml:3]: just out of curiosity, were these intentionally changed to double quotes and have a specific reason? i'm fine either way, wondering if there was an issue

**Review by jeff-dude — CHANGES_REQUESTED** [review]

File: models/curvefi/ethereum/curvefi_ethereum_sources.yml (hunk 2 above)
Comment (agaperste) [reply models/curvefi/ethereum/curvefi_ethereum_sources.yml]: added to "base_sources" model 
I did a different "freshness" as I was thinking contract potentially doesn't get new data as much as txns. But let me know if you feel differently and I will take out the freshness override.

File: dbt_project.yml (hunk 4 above)
Comment (agaperste) [reply dbt_project.yml:3]: I have dbt formatter extension installed on my VSCode, and it auto formats this to double quotes. I even tried to disable the formatter temporarily but somehow it still force formatted to double quotes whenever I saved 🫠, so I just left it be

File: models/curvefi/ethereum/curvefi_ethereum_view_pools.sql [hunk 5]
```diff
@@ -0,0 +1,285 @@
...
+        _symbol AS symbol,
+        output_0 AS pool_address,
+        "_A" AS A,
+        _fee AS mid_fee,
+        _fee AS out_fee,
+        output_0 AS token_address,
+        output_0 AS deposit_contract,
+        _coin AS coin0,
+        CASE
+            WHEN _base_pool = '0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7' THEN '0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490' --changing from swap to token contract
```
Comment (dot2dotseurat) [inline models/curvefi/ethereum/curvefi_ethereum_view_pools.sql:140]: I think setting these as variables would be a really good showcase of best practices for the wizards.

File: dbt_project.yml (hunk 4 above)
Comment (dot2dotseurat) [reply dbt_project.yml:3]: which one do you have? maybe we can all use it.

**Comment by dot2dotseurat** [comment]
@agaperste great work. really good job on the descriptions. my only nit is if we can move the addresses to dbt variables, I feel like it would be a good showcase of that capability.

File: models/curvefi/ethereum/curvefi_ethereum_sources.yml (hunk 2 above)
Comment (jeff-dude) [reply models/curvefi/ethereum/curvefi_ethereum_sources.yml]: yes, custom freshness is good when you know the decoded table refresh rate. we've had recent changes pushed to the repo around editing source freshness dependent on each decoded table

File: dbt_project.yml (hunk 4 above)
Comment (jeff-dude) [reply dbt_project.yml:3]: may be worth it for us all to look at internally (although could still get external user PRs which don't use it)

**Comment by jeff-dude** [comment]
happy to help build and test once changes are published via meghan's comments

**Comment by agaperste** [comment]
thanks for the comments! sorry for the delay, will address the comments by end of week!

File: models/curvefi/ethereum/curvefi_ethereum_view_pools.sql (hunk 5 above)
Comment (agaperste) [reply models/curvefi/ethereum/curvefi_ethereum_view_pools.sql]: converted to variables! good call!

**Comment by agaperste** [comment]
@jeff-dude Made changes according to comments! Please let me know if you need anything else from my side!

**Comment by soispoke** [comment]
Hi @agaperste :) 

I'm not 100% sure but I think @dot2dotseurat was thinking of using variables inside the models themselves, like here:
https://github.com/duneanalytics/spellbook/pull/1242/files/69cf1f302a8fddafa09699cb2148691bd4c625b5#diff-9133917a07d3bbcab0339a52d96a96e83c5a66a1275a9326adff5e6e3ca7bcd5

I don't have a strong opinion, but stacking variables in the dbt_project.yml file might end up being a bit long/messy

wdyt @jeff-dude ?

**Comment by agaperste** [comment]
@soispoke Thanks for flagging this! You are right probably good not to clog up dbt_project.yml

I have updated the PR accordingly @jeff-dude

**Comment by jeff-dude** [comment]
> Hi @agaperste :)
> 
> I'm not 100% sure but I think @dot2dotseurat was thinking of using variables inside the models themselves, like here: https://github.com/duneanalytics/spellbook/pull/1242/files/69cf1f302a8fddafa09699cb2148691bd4c625b5#diff-9133917a07d3bbcab0339a52d96a96e83c5a66a1275a9326adff5e6e3ca7bcd5
> 
> I don't have a strong opinion, but stacking variables in the dbt_project.yml file might end up being a bit long/messy
> 
> wdyt @jeff-dude ?

agree 💯

**Comment by jeff-dude** [comment]
thanks again @agaperste -- i've tested locally and the seeds/models build fine (using 'dbt_jeff_' preface on schema names if you want to query. all tests pass as well. at this point, only open question is the height task i created about seed file usage. once confirmed, we can finalize and merge this 👍

**Comment by agaperste** [comment]
@jeff-dude removed seed! very much looking forward to getting this merged!

**Review by jeff-dude — APPROVED** [review]
pushed a change to alias the new view model. otherwise, seeds & models build fine, all tests passed 👍 
fyi @soispoke

**Comment by agaperste** [comment]
thank you @jeff-dude!!

**Pull request merged**